apps/meteor/server/services/team/service.ts

```diff
@@ -633,14 +633,19 @@ export class TeamService extends ServiceClassInternal implements ITeamService {
 			};
 		}
 
-		const users = await Users.findActive({ ...query }).toArray();
-		const userIds = users.map((m) => m._id);
-		const cursor = TeamMember.findMembersInfoByTeamId(teamId, count, offset, {
-			userId: { $in: userIds },
-		});
+		const cursor = this.TeamMembersModel.findMembersInfoByTeamId(teamId, count, offset);
+		const records = await cursor.toArray();
+		const teamUserIds = records.map((m) => m.userId);
+
+		query._id = {
+			$in: teamUserIds,
+		};
```

Comment:
I think we should check if `query._id` is undefined here before assigning this new value so as to prioritize the user's query (if there is any)

Reply:
Updated!